feat: Release PR checklist action

premanshup opened the pull request:
### Description
<!-- Please describe what you have changed or added -->
This PR has an action which updates the PR when we add label `Release PR checklist` to the PR with base branch `master` or `dev` or `next-release`.

Title: `Build: Release PR`
Body:
```
### Description
   - Final release PR. Please check all the checklist below before merging.
   ---

   ### Checklist:
   - [ ] Finalize changelog.txt
   - [ ] Add doc links in the changelog.txt
   - [ ] Change minimum addon update version - ASTRA_EXT_MIN_VER
   - [ ] Run to update packages        : `npm audit fix`
   - [ ] Run to produce minified files : `grunt minify`
   - [ ] Run to update Google Fonts    : `grunt google-fonts`
   - [ ] PHPCS - vendor/bin/phpcs
   - [ ] Run to update version number  : `grunt version-bump --ver=<version-number>`
   - [ ] Check the Background Process function version
   - [ ] Update - Tested Upto - style.css & readme.txt
   - [ ] Update Readme.txt file and then run: `grunt readme`
   - [ ] Generate POT file             : `wp i18n make-pot wp-content/themes/astra wp-content/themes/astra/languages/astra.pot`
   - [ ] Add draft Release notes on github
   - [ ] Add draft changelogs to wpastra.com
   - [ ] Open PR to backport branch to `next-release`
```

### Screenshots
<!-- if applicable -->

#### Example PR - #2628 

#### Before:
![Screenshot at April 14th 2021 - 12 53 40 pm@1 25x](https://user-images.githubusercontent.com/36841355/114670298-7907ca00-9d20-11eb-861f-1c81d7229487.png)

#### After ( label `Release PR checklist` )
![Annotation on 2021-04-14 at 12-59-07](https://user-images.githubusercontent.com/36841355/114671015-40b4bb80-9d21-11eb-80b8-bb9a79ba537f.png)

### Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change -->
New feature for GitHub action.

### How has this been tested?
<!-- Please describe in detail how you tested your changes. -->

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included any necessary tests <!-- if applicable -->
- [x] I've included developer documentation <!-- if applicable -->
- [x] I've added proper labels to this pull request <!-- if applicable -->

vrundakansara commented:
Cool! 👍🏼

Nikschavan commented:
@premanshup if the person who created the pr had already added some description to the pr (like we usually include the changelog in the pr description) will that be removed and replaced with the checklist?

premanshup commented:
> @premanshup if the person who created the pr had already added some description to the pr (like we usually include the changelog in the pr description) will that be removed and replaced with the checklist?

Yes, it will be removed and replaced with the above checklist. Though there is an option to change it. Instead of 

```
body-update-action: 'replace'
title-update-action: 'replace'
```

We can do `prefix` or `suffix` :

```
body-update-action: 'suffix'
title-update-action: 'suffix'
```

This will append checklist to the description/title without deleting the existing description/title.

Please let me know if I should change replace with this?
